src/fmu/tools/utilities/sample_attributes.py

```diff
@@ -0,0 +1,162 @@
+            in xtgeo to achieve this.
+        attribute_error_minimum: The minimum error to apply to the attribute (optional).
+        region: The region parameter to sample from the grid (optional).
+        zone: The zone parameter to sample from the grid (optional).
+        position: The position to sample the attributes on the grid. This
+            shall be given as a tuple, as e.g. ("MyZone", "center") where the first
+            is zone name, and the second is position ("top", "center", "base") in
+            that zone. Default is ("", "center") which will take the middle layer
+            of the total grid. The zone name is case sensitive. If zone is not given,
+            the full grid interval will be applied to determine the layer.
+        **kwargs: Additional keywords (developer settings).
+
+    Returns:
+        pd.Dataframe: Points with the sampled attributes and attributes combined.
+    """
+    logger.info("Sampling attributes on grid resolution as points set.")
+
+    debug = kwargs.get("debug", False)
+
+    layer = _get_layer(position, grid, zone)
+    logger.debug("Layer index (base=1) to apply is: %s", layer)
+
+    template = xtgeo.surface_from_grid3d(
+        grid,
+        template="native",  # used to appromimate the native grid resolution
+        where=layer,
+        property="i",
+    )
+    # prepare the attribute and error to the grid
+    attribute_sampled = template.copy()
+    attribute_error_sampled = template.copy()
+
+    # do the resampling and get the points as dataframe
+    attribute_sampled.resample(attribute)
+    dataframe = _dataframe_from_surface(
+        attribute_sampled, newname=Attrs.OBS.value, debug=debug
+    )
+
+    if isinstance(attribute_error, float):
+        if attribute_error < 0:
+            raise ValueError("The attribute error shall be an absolute value.")
+        err = attribute * attribute_error
+        err.values = np.abs(err.values)
+    else:
+        err = attribute_error
+        if err.values.min() < 0:
+            raise ValueError("The attribute error shall be an absolute value.")
+
+    if attribute_error_minimum:
+        err.values = np.maximum(err.values, attribute_error_minimum)
+
+    attribute_error_sampled.resample(err)
+    df_err = _dataframe_from_surface(
+        attribute_error_sampled, newname=Attrs.OBS_ERROR.value, debug=debug
+    )
+    dataframe[Attrs.OBS_ERROR.value] = df_err[Attrs.OBS_ERROR.value]
+
+    df_region = None
+    if region:
+        region_sampled = xtgeo.surface_from_grid3d(
+            grid, template=template, where=layer, property=region
+        )
+        region_sampled.to_file("/tmp/region.gri")
+        df_region = _dataframe_from_surface(
+            region_sampled, newname=Attrs.REGION.value, debug=debug
+        )
+        dataframe[Attrs.REGION.value] = df_region[Attrs.REGION.value]
+
+    # remove lines without NaN values, if any
```

Comment:
don't need this comment I think 🙂 

```suggestion
    # remove lines with NaN values, if any
```